File: 1514-minimum-value-to-get-positive-step-by-step-sum/minimum-value-to-get-positive-step-by-step-sum.ts
function minStartValue(nums: number[]): number {
  if (!nums.filter((i) => i < 0).length) return 1;
  let sum = 0;
  let minSum = 0;
  for (let i = 0; i < nums.length; i++) {
    sum += nums[i];
    if (sum < minSum) {
      minSum = sum;
    }
  }
  return minSum * -1 + 1;
}

File: 1514-minimum-value-to-get-positive-step-by-step-sum/minimum-value-to-get-positive-step-by-step-sum.java
class Solution {
    public int minStartValue(int[] nums) {
 if (allPositive(nums)) {
      return 1;
    }
    int sum = 0;
    int minSum = 0;
    for (int i = 0; i < nums.length; i++) {
      sum += nums[i];
      minSum = Math.min(minSum, sum);
    }
    return minSum * -1 + 1;

    }
    private boolean allPositive(int[] nums) {
    for (int i = 0; i < nums.length; i++) {
      if (nums[i] < 0) {
        return false;
      }
    }
    return true;
  }
}